pkg/deploy/generator/generators.go

```diff
@@ -114,7 +114,13 @@ func (g *generator) Artifacts() error {
 }
 
 func (g *generator) writeTemplate(t *arm.Template, output string) error {
-	b, err := g.templateFixup(t)
+
+	sharedAccessKeyHack := false
+	if output == FileRPProduction {
```

Comment:
For Nont and Weinong:

Without this `if`, some other files besides `rp-production.json` get modified, such as `rp-oicd.json`, among others. 

As I am not sure yet which set of the files we want to modify, I added this to help us control that.

I'm thinking probably all files, as we don't want any more SAS keys anywhere. But, still, being able to control it feels safer - I don't know if all the storage accounts have other authentication ways and are ready for SAS to be disabled.